app/controllers/TaxYearController.scala

```diff
@@ -75,49 +75,62 @@ class TaxYearController @Inject()(val controllerComponents: MessagesControllerCo
-              case (Some(true), _, Some(versions)) => VersionInfo(versions, Compiled)
-              case (_, Some(true), Some(versions)) => VersionInfo(versions, Submitted)
-              case _ => VersionInfo(1, NotStarted)
-            }
+      eventReportingConnector.getFeatureToggle("lta-events-show-hide").flatMap { ltaToggle =>
+        val form = getFormProviderByLtaToggle(ltaToggle.isEnabled)
+        form.bindFromRequest().fold(
+          formWithErrors =>
+            eventReportingConnector.getFeatureToggle("event-reporting-tax-year").flatMap { taxYearToggle =>
+              Future.successful(renderPage(formWithErrors, waypoints, BadRequest, taxYearToggle.isEnabled))
+            },
+          value => {
+            val originalUserAnswers = request.userAnswers
+            val vd = originalUserAnswers
+              .get(EventReportingOverviewPage).toSeq.flatten.find(_.taxYear == value).flatMap(_.versionDetails)
+            val versionInfo =
+              (vd.map(_.compiledVersionAvailable), vd.map(_.submittedVersionAvailable), vd.map(_.numberOfVersions)) match {
+                case (Some(true), _, Some(versions)) => VersionInfo(versions, Compiled)
+                case (_, Some(true), Some(versions)) => VersionInfo(versions, Submitted)
+                case _ => VersionInfo(1, NotStarted)
+              }
 
-          val futureAfterClearDown = request.userAnswers.get(TaxYearPage) match {
-            case Some(v) if v != value => userAnswersCacheConnector.removeAll(request.pstr)
-            case _ => Future.successful((): Unit)
-          }
+            val futureAfterClearDown = request.userAnswers.get(TaxYearPage) match {
+              case Some(v) if v != value => userAnswersCacheConnector.removeAll(request.pstr)
+              case _ => Future.successful((): Unit)
+            }
 
-          val updatedAnswers = originalUserAnswers
-            .setOrException(TaxYearPage, value, nonEventTypeData = true)
-            .setOrException(VersionInfoPage, versionInfo, nonEventTypeData = true)
-          futureAfterClearDown.flatMap { _ =>
-            userAnswersCacheConnector.save(request.pstr, updatedAnswers).map { _ =>
-              Redirect(TaxYearPage.navigate(waypoints, originalUserAnswers, updatedAnswers).route)
+            val updatedAnswers = originalUserAnswers
+              .setOrException(TaxYearPage, value, nonEventTypeData = true)
+              .setOrException(VersionInfoPage, versionInfo, nonEventTypeData = true)
+            futureAfterClearDown.flatMap { _ =>
+              userAnswersCacheConnector.save(request.pstr, updatedAnswers).map { _ =>
+                Redirect(TaxYearPage.navigate(waypoints, originalUserAnswers, updatedAnswers).route)
+              }
             }
           }
-        }
-      )
+        )
+      }
   }
 
+  private def getFormProviderByLtaToggle(isEnabled: Boolean): Form[TaxYear] = {
```

Comment:
Maybe add `TODO` to track removal of this form provider in future

Reply:
added TODO comment to removal in future